src/services/token/readOnlyTokenHook.ts

```diff
@@ -0,0 +1,52 @@
+/**
+ *  TODO: Extract this code out to its own package to reuse in tradetrust-website
+ */
+
+import { ReadOnlyToken } from "@govtechsg/oa-token";
+
+import { getLogger } from "../../utils/logger";
+
+import { useState, useEffect, useCallback } from "react";
+import { WrappedDocument } from "@govtechsg/open-attestation";
+import { useEthereumTransactionState, TransactionState } from "./helpers";
+const { trace, error } = getLogger("useToken");
+
+type UseReadOnlyToken = { state: TransactionState; tokenInstance: ReadOnlyToken | null; getTokenOwner: GetTokenOwner };
+type GetTokenOwner = () => Promise<string>;
+
+export const useReadOnlyToken = ({ document }: { document: WrappedDocument<any> }): UseReadOnlyToken => {
```

Comment:
i suspect it would be tidier to name this useTokenInfo() and handle reading of all meta in here, e.g 

- token registry
- token issuance status
- token owner if issued
- token owner is EOA or TitleEscrow

then we have
const [status, beneficiary address, holder address] = useTitleEscrow({owner address})

and then we handle the permutation of states

const [status, transferTo] = useTransferHoldership({TitleEscrow instance})
etc...